tests/AzureBlobStorageAdapterTest.php

```diff
@@ -0,0 +1,480 @@
+<?php
+declare(strict_types=1);
+
+namespace MedTrainer\Flysystem\AzureBlobStorage\Tests;
+
+use Exception;
+use League\Flysystem\Config;
+use League\Flysystem\FileAttributes;
+use League\Flysystem\UnableToCopyFile;
+use League\Flysystem\UnableToCreateDirectory;
+use MedTrainer\Flysystem\AzureBlobStorage\AzureBlobStorageAdapter;
+use MicrosoftAzure\Storage\Blob\BlobRestProxy;
+use MicrosoftAzure\Storage\Blob\Models\GetBlobPropertiesResult;
+use MicrosoftAzure\Storage\Blob\Models\GetBlobResult;
+use MicrosoftAzure\Storage\Blob\Models\ListBlobsResult;
+use MicrosoftAzure\Storage\Blob\Models\PutBlobResult;
+use MicrosoftAzure\Storage\Common\Exceptions\ServiceException;
+use PHPUnit\Framework\MockObject\MockObject;
+use PHPUnit\Framework\TestCase;
+use Psr\Http\Message\ResponseInterface;
+use Psr\Log\LoggerInterface;
+
+class AzureBlobStorageAdapterTest extends TestCase
+{
+    const FILE_TEST = 'test_image.png';
+
+    const DIR_TEST = 'mt';
+
+    /** @var MockObject|LoggerInterface  */
+    private $logger;
+
+    public function setUp(): void
+    {
+        $this->logger = $this->createMock(LoggerInterface::class);
+        $this->logger->expects(self::any())
```

Comment:
can we change this to expect the logger to be called at least once?
```suggestion
        $this->logger->expects(self::atLeastOnce())
```